.azure-pipelines/pipelines.yml

```diff
@@ -286,11 +286,12 @@ stages:
         AZP_BRANCH: $(Build.SourceBranch)
 
 - stage: linux_x64
-  dependsOn: ["precheck"]
+  dependsOn: ["env", "precheck"]
   # For post-submit builds, continue even if precheck fails
   condition: and(not(canceled()), or(succeeded(), eq(variables['PostSubmit'], true)))
   jobs:
   - job: release
+    condition: and(not(canceled()), succeeded(), ne(stageDependencies.env.repo.outputs['changed.mobileOnly'], 'true'), ne(stageDependencies.env.repo.outputs['changed.docsOnly'], 'true'))
```

Comment:
probably we can remove the `not(canceled())` logic here as that is handled by the stage condition

Reply:
i have removed the succeded condition as that is handled above but left the not canceled as we defo dont want it to run in that case